devdoc/EndUsers.md

```diff
@@ -21,18 +122,100 @@ Welcome to the End User Documentation. This guide provides a comprehensive colle
 [Options for Embedded Java in HPCC](../plugins/javaembed/javaembedOptions.md)
 This article describes configuration options for the embedded Java plugin in HPCC Platform, which allows ECL code to execute Java functionality.
 
-## EVTool
+## ESDL
+
+Comprehensive instructions for working with Enterprise Services Definition Language (ESDL) and the ESDL Tools.
 
-## FTSlave
+[ESDL ReadMe](../esp/esdllib/README.md)
+General documentation for ESDL library.
 
-## ESDL
+[ESDL Command Line Tool](../tools/esdlcmd/README.md)
+Documentation for ESDL command-line tool.
+
+[ESP API tool](../tools/esp-api/README.md)
+Documentation for ESP API tool.
+
+[ESDL Security Details](../esp/esdllib/README-SECURITY.md)  
+Security-related documentation for ESDL library.
+
+[ESDL functions](ESDLFunctions.md) List of ESDL Functions with links to details.
 
 ## Security
 
+[Security User Authentication](SecurityUserAuthentication.md) User authentication mechanisms and configuration.
+
+[Security Configuration](SecurityConfig.md) Configuration options for HPCC Systems Platform security.
+
+[JWT Security plugin](../system/security/plugins/jwtSecurity/README.md) Documentation for JWT Security plugin.
+
 ### LDAP
 
 ### Secrets
 
+Best practices for managing secrets securely within the system.
+
+- [Kubernetes Secrets](../helm/examples/secrets/README-kubernetessecrets.md)
+
+- [Vault Secrets K8s Authentication](../helm/examples/secrets/README-vault_secrets_using_kubernetes_authentication.md)
+
+- [Vault Secrets Client Cert Vault Authentication](../helm/examples/secrets/README-vaultsecretsusingclientcertvaultauthentication.md)
+
 ## Roxie
 
+Roxie is the query processing engine, optimized for real-time data delivery. It is designed to handle high-concurrency, low-latency queries efficiently, making it ideal for queries requiring rapid responses.
+
+[Roxie Technical Details](roxie.md) Technical details and usage of the Roxie engine.
+
+[Roxie FAQ](./userdoc/roxie/FAQ.md) Frequently asked questions about Roxie.
+
+[Optimizing Roxie Query Performance](OptimizingRoxieQueryPerformance.md) Explains how to configure Roxie to maintain consistent query response times by setting minimum execution thresholds through various methods including configuration files, ECL code options, and URL parameters. It also describes how to monitor performance.
+
+[Memory Manager](MemoryManager.md) Documentation on memory management strategies.
+
 ## Thor
+
+## Plugins
+
```

Comment:
remove blank line for consistency

Reply:
Headings should be surrounded by blank lines.